### PR TITLE
feat(dashboard,web): opt-in app switching and redirects

### PR DESCRIPTION
**Pull request by ChmaraX**
### What changed? Why was the change needed?

- added providers to check if user is opted-in/out, in order to redirect:
  - if user is opted-in and visits legacy dashboard path which exists in new dashboard - redirect there
  - if user is opted-out and visits new dashboard - redirect back to old
- automatically set theme to light when user opts in for new dashboard

### CODE REVIEW

File: apps/dashboard/src/context/opt-in-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { useNavigate } from 'react-router-dom';
+
+export const OptInProvider = (props: PropsWithChildren) => {
+  const navigate = useNavigate();
+  const { children } = props;
+  const { status, isLoaded } = useNewDashboardOptIn();
+
+  useEffect(() => {
+    if (isLoaded && status !== NewDashboardOptInStatusEnum.OPTED_IN) {
+      window.location.href = '/legacy/workflows';
```
Comment (ChmaraX) [inline apps/dashboard/src/context/opt-in-provider.tsx]: using `navigate()` here won't work since its external url

File: apps/web/src/hooks/useNewDashboardOptIn.ts [hunk 2]
```diff
@@ -28,9 +29,11 @@ export function useNewDashboardOptIn() {
     window.location.href = NEW_DASHBOARD_URL || window.location.origin;
   };
 
-  const optIn = () => {
+  const optIn = async () => {
     segment.track('New dashboard opt-in');
-    updateUserOptInStatus(NewDashboardOptInStatusEnum.OPTED_IN);
+    await updateUserOptInStatus(NewDashboardOptInStatusEnum.OPTED_IN);
+    localStorage.setItem('mantine-theme', 'light');
```
Comment (ChmaraX) [inline apps/web/src/hooks/useNewDashboardOptIn.ts:35]: this needs to be set directly in local storage otherwise when using hook the operation seems to be async and can't be awaited

File: apps/web/src/components/providers/OptInProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+       * - /legacy/workflows/edit/123 -> /workflows
+       */
+      if (NEW_DASHBOARD_ROUTES.some((route) => currentRoute.includes(route))) {
+        /**
+         * TODO: in order to redirect to the same route, we need to translate the
+         * "dev_env_<id>" or wf/step slugs to legacy environment id and vice-versa
+         *
+         * note: /legacy is part of public URL, so we can't navigate() outside of that
+         */
+        window.location.href = window.location.origin;
```
Comment (ChmaraX) [inline apps/web/src/components/providers/OptInProvider.tsx:43]: for now we redirect to the default page which is the workflows page - otherwise we would need to parse old env id to new env slugs to redirect to `env/<env_slug>/workflows` - will do in separate PR if there is a hard requirement to keep env id

**Review by SokratisVidros — APPROVED** [review]
